### PR TITLE
Switch the project to es2015 module

### PR DESCRIPTION
**Pull request by jeremierobert-coveo**
# What was changed ?

I changed configuration files to enable webpack **tree-shaking**.
The goal of this is to ensure that when the project is imported somewhere, only required components will affect the **global scope** instead of the whole library.
I also change the file structure a bit to ease this change.

# Was was tested ?

I checked each task within the `package.json` file

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,16 +2,20 @@
...
+
+    "main": "./bin/js/index.js",
+    "types": "./bin/typings",
+
+    "sideEffects": false,
+    
     "scripts": {
-        "test": "karma start --single-run",
+        "test": "karma start ./config/karma.config.js --single-run",
         "testCoverage": "npm run test && cat ./bin/coverage/lcov.info | node ./node_modules/coveralls/bin/coveralls.js",
```
Comment (mikegron) [inline package.json]: You need to change the path here for `../node_modules/`,  I think that's why coveralls didn't work?

**Comment by mikegron** [comment]
Nice, I like the new file structure a lot better!

File: package.json (hunk 1 above)
Comment (jeremierobert-coveo) [reply package.json]: Will try but it does not makes sense to me. I mean, the package.json file is on root 😕 .

Comment (mikegron) [reply package.json]: oh you're right.. nevermind then :P

Comment (mikegron) [reply package.json]: ah, make sure the coverage is output in the right folder, in the karma.config.json, it outputs to `./bin/coverage`, pretty sure that will be in the config folder then

Comment (jeremierobert-coveo) [reply package.json]: Oups 😱

**Pull request merged**